### PR TITLE
Frontend: Add account statement export button

### PR DESCRIPTION
**Pull request by bisgardo**
## Purpose

Expose account statement export (#15) as a button in the frontend.

## Changes

Add an "Export" button in the account export widget.

*Before:*

image

*After:*

image

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.

### CODE REVIEW

**Comment by jens-concordium** [comment]
Approved from a design perspective.

**Review by soerenbf — COMMENTED** [review]
Couldn't sucessfully export an account statement. I assume this doesn't have anything to do with these changes, but shouldn't this work on stagenet currently?
Screenshot 2022-11-02 at 14 35 25

**Comment by bisgardo** [comment]
> Couldn't sucessfully export an account statement. I assume this doesn't have anything to do with these changes, but shouldn't this work on stagenet currently? Screenshot 2022-11-02 at 14 35 25

There is an extra `graphql` in the URL that shouldn't be there (note the `..` in `${apiUrl}/../rest/export/statement?accountAddress=${props.account.address.asString}`). Is this the URL that the frontend invokes or just a test?

**Comment by soerenbf** [comment]
> There is an extra `graphql` in the URL that shouldn't be there (note the `..` in `${apiUrl}/../rest/export/statement?accountAddress=${props.account.address.asString}`). Is this the URL that the frontend invokes or just a test?

I just clicked the export button 😄

**Comment by bisgardo** [comment]
> > There is an extra `graphql` in the URL that shouldn't be there (note the `..` in `${apiUrl}/../rest/export/statement?accountAddress=${props.account.address.asString}`). Is this the URL that the frontend invokes or just a test?
> 
> I just clicked the export button smile
> 
> It it a vue thing that it can resolve a url like that, or is it native anchor href stuff? Never used it before. Alternatively you could pull in something to resolve a constructed path (like nodejs path.resolve(apiUrl, '..', 'rest/export/statement'), which I think should work.

Christ. It happened because there was an extra `/` on the URLs of the non-dev envs. Will try and use such a path resolver if I can do it without adding a new dependency.

**Comment by soerenbf** [comment]
> Christ. It happened because there was an extra `/` on the URLs of the non-dev envs. Fixed them. Will try and use such a path resolver if I can do it without adding a new dependency.

I would be surprised if you could 😀 

> Regarding if it's a vue thing, no it's just standard string interpolation.

I meant the `../` part of the url. Was just surprised that it could be a part of a valid URL used in a href attribute 😉

**Comment by bisgardo** [comment]
> > Christ. It happened because there was an extra `/` on the URLs of the non-dev envs. Fixed them. Will try and use such a path resolver if I can do it without adding a new dependency.
> 
> I would be surprised if you could grinning

It actually turns out that you can :exploding_head: 

> > Regarding if it's a vue thing, no it's just standard string interpolation.
> 
> I meant the `../` part of the url. Was just surprised that it could be a part of a valid URL used in a href attribute wink

Ah yeah that's just like file paths (which is what URL paths originally was). Except that `//../` is then handled differently than (at least Linux) does it :palm_tree:

File: frontend/src/components/Accounts/AccountDetailsAccountStatement.vue [hunk 1]
```diff
@@ -7,7 +7,9 @@
 					<TableTh>Type</TableTh>
 					<TableTh v-if="breakpoint >= Breakpoint.XL">Reference</TableTh>
 					<TableTh align="right">Amount</TableTh>
-					<TableTh v-if="breakpoint >= Breakpoint.XXL" align="right">Balance</TableTh>
+					<TableTh v-if="breakpoint >= Breakpoint.XXL" align="right"
+						>Balance</TableTh
+					>
```
Comment (soerenbf) [inline frontend/src/components/Accounts/AccountDetailsAccountStatement.vue:12]: What happened here? that formatting is a bit messed up, is it not? 👀

File: frontend/src/components/Accounts/AccountDetailsContent.vue [hunk 2]
```diff
@@ -201,8 +201,14 @@ type Props = {
...
 
 const { apiUrl } = useRuntimeConfig()
-const exportUrl = `${apiUrl}/../rest/export/statement?accountAddress=${props.account.address.asString}`
+
+function exportUrl(accountAddress) {
+	const url = new URL(apiUrl)
+	url.pathname = 'rest/export/statement' // setting pathname discards any existing path in 'apiUrl'
+	url.searchParams.append('accountAddress', accountAddress)
+	return url.toString()
+}
```
Comment (soerenbf) [inline frontend/src/components/Accounts/AccountDetailsContent.vue:213]: 🤯

**Review by soerenbf — APPROVED** [review]

File: frontend/src/components/Accounts/AccountDetailsAccountStatement.vue (hunk 1 above)
Comment (bisgardo) [reply frontend/src/components/Accounts/AccountDetailsAccountStatement.vue:12]: I thought so too so fixed it, but the formatter reverted it :shrug:

File: frontend/src/components/Accounts/AccountDetailsContent.vue (hunk 2 above)
Comment (bisgardo) [reply frontend/src/components/Accounts/AccountDetailsContent.vue:213]: I know!

**Pull request merged**